web/api/queries/query.c

```diff
@@ -1510,32 +1517,32 @@ static void rrd2rrdr_log_request_response_metadata(RRDR *r
         , RRDR_OPTIONS options __maybe_unused
         , RRDR_GROUPING group_method
         , bool aligned
-        , long group
-        , long resampling_time
-        , long resampling_group
+        , size_t group
+        , time_t resampling_time
+        , size_t resampling_group
         , time_t after_wanted
         , time_t after_requested
         , time_t before_wanted
         , time_t before_requested
-        , long points_requested
-        , long points_wanted
+        , size_t points_requested
+        , size_t points_wanted
         //, size_t after_slot
         //, size_t before_slot
         , const char *msg
         ) {
 
-    time_t first_entry_t = rrdset_first_entry_t(r->st);
-    time_t last_entry_t = rrdset_last_entry_t(r->st);
+    time_t first_entry_t = r->internal.qt->db.first_time_t;
+    time_t last_entry_t = r->internal.qt->db.last_time_t;
 
-    info("INTERNAL ERROR: rrd2rrdr() on %s update every %d with %s grouping %s (group: %ld, resampling_time: %ld, resampling_group: %ld), "
+    info("INTERNAL ERROR: rrd2rrdr() on %s update every %ld with %s grouping %s (group: %ld, resampling_time: %ld, resampling_group: %ld), "
```

Comment:
During compilation I observed different warnings on x86_64 with this PR, so I am writing the now:

```sh
In file included from ./libnetdata/libnetdata.h:433,
                 from ./daemon/common.h:6,
                 from ./web/api/web_api_v1.h:6,
                 from ./web/api/formatters/rrd2json.h:6,
                 from web/api/queries/query.c:4:
web/api/queries/query.c: In function 'rrd2rrdr_log_request_response_metadata':
web/api/queries/query.c:1537:10: warning: format '%ld' expects argument of type 'long int', but argument 9 has type 'size_t' {aka 'long unsigned int'} [-Wformat=]
 1537 |     info("INTERNAL ERROR: rrd2rrdr() on %s update every %ld with %s grouping %s (group: %ld, resampling_time: %ld, resampling_group: %ld), "
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
......
 1550 |          , group
      |            ~~~~~
      |            |
      |            size_t {aka long unsigned int}
./libnetdata/log/log.h:98:71: note: in definition of macro 'info'
   98 | #define info(args...)    info_int(__FILE__, __FUNCTION__, __LINE__, ##args)
      |                                                                       ^~~~
web/api/queries/query.c:1537:91: note: format string is defined here
 1537 |     info("INTERNAL ERROR: rrd2rrdr() on %s update every %ld with %s grouping %s (group: %ld, resampling_time: %ld, resampling_group: %ld), "
      |                                                                                         ~~^
      |                                                                                           |
      |                                                                                           long int
      |                                                                                         %ld
web/api/queries/query.c:1537:10: warning: format '%ld' expects argument of type 'long int', but argument 11 has type 'size_t' {aka 'long unsigned int'} [-Wformat=]
 1537 |     info("INTERNAL ERROR: rrd2rrdr() on %s update every %ld with %s grouping %s (group: %ld, resampling_time: %ld, resampling_group: %ld), "
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
......
 1552 |          , resampling_group
      |            ~~~~~~~~~~~~~~~~
      |            |
      |            size_t {aka long unsigned int}
./libnetdata/log/log.h:98:71: note: in definition of macro 'info'
   98 | #define info(args...)    info_int(__FILE__, __FUNCTION__, __LINE__, ##args)
      |                                                                       ^~~~
web/api/queries/query.c:1537:136: note: format string is defined here
 1537 |     info("INTERNAL ERROR: rrd2rrdr() on %s update every %ld with %s grouping %s (group: %ld, resampling_time: %ld, resampling_group: %ld), "
      |                                                                                                                                      ~~^
      |                                                                                                                                        |
      |                                                                                                                                        long int
      |                                                                                                                                      %ld
web/api/queries/query.c:1537:10: warning: format '%ld' expects argument of type 'long int', but argument 24 has type 'size_t' {aka 'long unsigned int'} [-Wformat=]
 1537 |     info("INTERNAL ERROR: rrd2rrdr() on %s update every %ld with %s grouping %s (group: %ld, resampling_time: %ld, resampling_group: %ld), "
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
......
 1580 |          , r->rows
      |            ~~~~~~~
      |             |
      |             size_t {aka long unsigned int}
web/api/queries/query.c:1537:10: warning: format '%ld' expects argument of type 'long int', but argument 25 has type 'size_t' {aka 'long unsigned int'} [-Wformat=]
 1537 |     info("INTERNAL ERROR: rrd2rrdr() on %s update every %ld with %s grouping %s (group: %ld, resampling_time: %ld, resampling_group: %ld), "
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
......
 1581 |          , points_wanted
      |            ~~~~~~~~~~~~~
      |            |
      |            size_t {aka long unsigned int}
./libnetdata/log/log.h:98:71: note: in definition of macro 'info'
   98 | #define info(args...)    info_int(__FILE__, __FUNCTION__, __LINE__, ##args)
      |                                                                       ^~~~
web/api/queries/query.c:1537:10: warning: format '%ld' expects argument of type 'long int', but argument 26 has type 'size_t' {aka 'long unsigned int'} [-Wformat=]
 1537 |     info("INTERNAL ERROR: rrd2rrdr() on %s update every %ld with %s grouping %s (group: %ld, resampling_time: %ld, resampling_group: %ld), "
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
......
 1582 |          , points_requested
      |            ~~~~~~~~~~~~~~~~
      |            |
      |            size_t {aka long unsigned int}

```

Reply:
solved